tlair/src/main/scala/at/forsyte/apalache/tla/lir/TlaType1.scala

```diff
@@ -98,19 +98,69 @@ case object StrT1 extends TlaType1 {
 }
 
 /**
- * An uninterpreted type constant.
+ * <p>An uninterpreted type constant such as PROC_NAME.</p>
+ *
+ * <p>Inside the type checker, this class may also represent references to type aliases. Do not use this feature outside
+ * of the type checker.</p>
  *
  * @param name
  *   unique name of the constant type
  */
 case class ConstT1(name: String) extends TlaType1 {
-  require(name.forall(c => c.isUpper || c.isDigit || c == '_'),
-      "ConstT1 accepts identifiers in upper case, found: " + name)
+  require(ConstT1.isUninterpreted(name) || ConstT1.isAliasReference(name),
+      "ConstT1 accepts identifiers in upper case or $aliasReference, found: " + name)
 
   override def toString: String = name
 
   override def usedNames: Set[Int] = Set.empty
+}
+
+/**
+ * A companion object for [[ConstT1]].
+ */
+object ConstT1 {
+
+  /**
+   * Does this type represent an uninterpreted type, e.g., PROCESS. Outside of the type parser and the type checker,
+   * this method should always return true.
+   *
+   * @param name
+   *   type name
+   * @return
+   *   true iff the type name represents an uninterpreted type.
+   */
+  def isUninterpreted(name: String): Boolean = {
+    name.forall(c => c.isUpper || c.isDigit || c == '_')
```

Comment:
Why don't you use regex here?

Reply:
Also, this is incorrect, because you can start your value name with `[0-9]` and the forall test would pass.
The regex is `[A-Z_][A-Z0-9_]` (see `ModelValueHandler.matchRegex`)

Reply:
I have opened a follow-up issue: https://github.com/informalsystems/apalache/issues/1992#issue-1307695067